gossip/comm/comm_impl.go

```diff
@@ -37,6 +37,10 @@ const (
 	DefSendBuffSize  = 20
 )
 
+var (
+	errProbe = errors.New("probe")
```

Comment:
Can't we have a more descriptive error message for better readability? 

Correct me if I have misunderstood: this error would be returned when the peer tries to authenticate a stream which is not a probe (by passing an appropriate flag) but it ends up being a probe. As a result, we don't store info in the connection store. If my understanding is correct, I am not sure whether it should be an error. Maybe a bool can be used instead as I mentioned in the other comment.

Reply:
You misunderstood. This error just means that the remote peer probed us.

Reply:
Okay. That's what I meant may be my wordings didn't convey that. That's why I suggested `isProbe bool` later in the comment.

Reply:
I see a function that returns an object and an error in the following way:
If the error is nil, use the first return value.
Else, use the other return value.

This perfectly fits my needs here.